packages/agent-utils/src/memory/Vectorizer.ts

```diff
@@ -0,0 +1,39 @@
+import { CreateEmbeddingResponse, OpenAIApi } from "openai";
+
+export class Vectorizer {
+  constructor(private readonly config: { api: OpenAIApi }) { }
```

Comment:
Do we need config for a single dependency?

Reply:
Refactoring this into a more general struct